LayoutKit/Layouts/TextViewLayout.swift

```diff
@@ -0,0 +1,162 @@
+                            configure: ((TextView) -> Void)? = nil) {
+        self.init(text: .unattributed(text),
+                  font: font,
+                  lineFragmentPadding: lineFragmentPadding,
+                  contentInset: contentInset,
+                  layoutMargins: layoutMargins,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    public convenience init(attributedText: NSAttributedString,
+                            font: UIFont = defaultFont,
+                            lineFragmentPadding: CGFloat = 0,
+                            contentInset: UIEdgeInsets = .zero,
+                            layoutMargins: UIEdgeInsets = .zero,
+                            textContainerInset: UIEdgeInsets = .zero,
+                            layoutAlignment: Alignment = defaultAlignment,
+                            flexibility: Flexibility = defaultFlexibility,
+                            viewReuseId: String? = nil,
+                            configure: ((TextView) -> Void)? = nil) {
+        self.init(text: .attributed(attributedText),
+                  font: font,
+                  lineFragmentPadding: lineFragmentPadding,
+                  contentInset: contentInset,
+                  layoutMargins: layoutMargins,
+                  textContainerInset: textContainerInset,
+                  layoutAlignment: layoutAlignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  configure: configure)
+    }
+
+    // MARK: - Layout protocol
+
+    open func measurement(within maxSize: CGSize) -> LayoutMeasurement {
+        let fittedSize = textSize(within: maxSize)
+        return LayoutMeasurement(layout: self, size: fittedSize.decreasedToSize(maxSize), maxSize: maxSize, sublayouts: [])
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    // MARK: - private helpers
+
+    private func textSize(within maxSize: CGSize) -> CGSize {
+        var size = TextCalculatorHelper.textSize(maxSize, text, font)
+
+        let heightInset = (textContainerInset.top + textContainerInset.bottom) + (layoutMargins.top + layoutMargins.bottom)
+        let widthInset = (textContainerInset.left + textContainerInset.right) + (layoutMargins.left + layoutMargins.right) + lineFragmentPadding * 2
+            + (contentInset.left + contentInset.right)
+        size = CGSize(width: size.width + widthInset, height: size.height + heightInset)
+
+        return size
+    }
+
+    // MARK: - overriden methods
+
+    open override func configure(view textView: TextView) {
```

Comment:
it'd be nice to add a comment that if you're trying to config the textContainerInset, contentInset, font, layoutMargins, etc that won't work and you'll need to set them on the class itself.